sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_service_client.hpp

```diff
@@ -104,5 +103,7 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
   private:
     Azure::Core::Url m_serviceUrl;
     std::shared_ptr<Azure::Core::Http::_internal::HttpPipeline> m_pipeline;
+
+    friend class ListSharesPagedResponse;
```

Comment:
Why is this friend relationship necessary?

Reply:
so that `ListSharesPagedResponse::OnNextPage` can access `m_serviceUrl` and `m_pipeline` to send HTTP requests.

Reply:
Please see my earlier comments about this.